src/plugins/kibana_react/public/util/mount_point_portal.test.tsx

```diff
@@ -0,0 +1,190 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FC } from 'react';
+import { mount, ReactWrapper } from 'enzyme';
+import { MountPoint, UnmountCallback } from 'kibana/public';
+import { MountPointPortal } from './mount_point_portal';
+import { act } from 'react-dom/test-utils';
+
+describe('MountPointPortal', () => {
+  let portalTarget: HTMLElement;
+  let mountPoint: MountPoint;
+  let setMountPoint: jest.Mock<(mountPoint: MountPoint<HTMLElement>) => void>;
+  let dom: ReactWrapper;
+
+  const refresh = () => {
+    new Promise(async (resolve) => {
+      if (dom) {
+        act(() => {
+          dom.update();
+        });
+      }
+      setImmediate(() => resolve(dom)); // flushes any pending promises
+    });
+  };
+
+  beforeEach(() => {
+    portalTarget = document.createElement('div');
+    document.body.append(portalTarget);
+    setMountPoint = jest.fn().mockImplementation((mp) => (mountPoint = mp));
+  });
+
+  afterEach(() => {
+    if (portalTarget) {
+      portalTarget.remove();
+    }
+  });
+
+  it('calls the provided `setMountPoint` during render', async () => {
+    dom = mount(
+      <MountPointPortal setMountPoint={setMountPoint}>
+        <span>portal content</span>
+      </MountPointPortal>
+    );
+
+    await refresh();
+
+    expect(setMountPoint).toHaveBeenCalledTimes(1);
+  });
+
+  it('renders the portal content when calling the mountPoint ', async () => {
+    dom = mount(
+      <MountPointPortal setMountPoint={setMountPoint}>
+        <span>portal content</span>
+      </MountPointPortal>
+    );
+
+    await refresh();
+
+    expect(mountPoint).toBeDefined();
+
+    act(() => {
+      mountPoint(portalTarget);
+    });
+
+    await refresh();
+
+    expect(portalTarget.textContent).toBe('portal content');
+  });
+
+  it('cleanup the portal content when the component is unmounted', async () => {
+    dom = mount(
+      <MountPointPortal setMountPoint={setMountPoint}>
+        <span>portal content</span>
+      </MountPointPortal>
+    );
+
+    act(() => {
+      mountPoint(portalTarget);
+    });
+
+    await refresh();
+
+    expect(portalTarget.textContent).toBe('portal content');
+
+    dom.unmount();
+
+    await refresh();
+
+    expect(portalTarget.textContent).toBe('');
```

Comment:
nit: `innerHTML` may be a better property to test against since `textContent` can be an empty string even if there are other DOM node children